File: movie-client/src/main/java/top/tonydon/util/JSONUtils.java
package top.tonydon.util;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;

public class JSONUtils {

    private static final ObjectMapper MAPPER;

    static {
        MAPPER = new ObjectMapper();
        MAPPER.registerModule(new JavaTimeModule());
    }

    public static <T> T parse(String json, Class<T> clazz) {
        try {
            return MAPPER.readValue(json, clazz);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
    }

    public static String toJSONStr(Object obj) {
        try {
            return MAPPER.writeValueAsString(obj);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
    }
}

File: client/src/main/java/top/tonydon/syncplayer/util/JSONUtils.kt
package top.tonydon.syncplayer.util

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule

object JSONUtils {
    private val MAPPER: ObjectMapper =  ObjectMapper()

    init {
        MAPPER.registerModule(JavaTimeModule())
    }

    fun <T> parse(json: String?, clazz: Class<T>?): T {
        return try {
            MAPPER.readValue(json, clazz)
        } catch (e: JsonProcessingException) {
            e.printStackTrace()
            throw RuntimeException(e)
        }
    }

    fun toJSONStr(obj: Any?): String {
        return try {
            MAPPER.writeValueAsString(obj)
        } catch (e: JsonProcessingException) {
            e.printStackTrace()
            throw RuntimeException(e)
        }
    }
}
